templates/homebrew.mxcl.buildkite-agent.plist.j2

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>Label</key>
+  <string>homebrew.mxcl.buildkite-agent</string>
+
+  <key>WorkingDirectory</key>
+  <string>/usr/local/bin</string>
+
+  <key>ProgramArguments</key>
+  <array>
+    <string>{{ buildkite_agent_executable[ansible_os_family] }}</string>
+    <string>start</string>
+    <string>--config</string>
+    <string>{{ buildkite_agent_conf_dir[ansible_os_family] }}/buildkite-agent.cfg</string>
+    <string>--spawn={{ buildkite_agent_count }}</string>
```

Comment:
Would you mind using an environment variable `BUILDKITE_AGENT_SPAWN` instead? That would make it more compatible without our own internal setup.

(Our own internal setup is a little involved, we actually made a script wrapper around bka start which first deals with some health assertions, makes sure our secrets are fetched at boot, etc)

Reply:
I don't see any other reference to environment variables so I'm not sure what you mean here? Do you want me to introduce a new configuration parameter and default to this environment variable instead of using `buildkite_agent_count`?

Just using the environment variable as is would require the user to specify it each time they run the playbook instead of being able to configure a var, seems more cumbersome to me.

This also realtes a bit to the CR about `when`.

I think this would be nice: `buildkite_agent_spawn: lookup('env', 'BUILDKITE_AGENT_SPAWN')` + `when: buildkite_agent_spawn is defined`.

WDYT?

Reply:
Sorry, I could have been clearer. I meant, add a new `BUILDKITE_AGENT_SPAWN` environment variable to this plist, and set its value to `{{ buildkite_agent_count | default('1') }}`.

(buildkite-agent will take configuration from the first place it finds, flag first, env-vars second, config-file last)